registration-processor/pre-processor/registration-processor-packet-classifier-stage/src/test/java/io/mosip/registration/processor/stages/packetclassifier/tagging/impl/ExceptionBiometricsTagGeneratorTest.java

```diff
@@ -1,123 +1,205 @@
+
+/**
+ * The Class PacketValidatorStageTest.
+ */
+@RefreshScope
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore({ "javax.management.*", "javax.net.ssl.*", "com.sun.org.apache.xerces.*", 
+	"javax.xml.*", "org.xml.*" })
+public class ExceptionBiometricsTagGeneratorTest {
+
+	private static String tagName = "EXCEPTION_BIOMETRICS";
+
+	private static String notAvailableTagValue = "--TAG_VALUE_NOT_AVAILABLE--";
+
+
+	private static String regClientCurrentVersion = "1.2.1-SNAPSHOT";
+
+	@InjectMocks
+	private ExceptionBiometricsTagGenerator exceptionBiometricsTagGenerator;
+
+	@Mock
+	ObjectMapper mapper;
+
+	Map<String, String> metamap = new HashMap<>();
+
+	@Before
+	public void setup() throws Exception {
+		Whitebox.setInternalState(exceptionBiometricsTagGenerator, "tagName", tagName);
+		Map<String, String> bioValueMapping = new LinkedHashMap<>();
+		bioValueMapping.put("leftLittle", "LL");
+		bioValueMapping.put("leftRing", "LR");
+		bioValueMapping.put("leftMiddle", "LM");
+		bioValueMapping.put("leftIndex", "LI");
+		bioValueMapping.put("leftThumb", "LT");
+		bioValueMapping.put("rightLittle", "RL");
+		bioValueMapping.put("rightRing", "RR");
+		bioValueMapping.put("rightMiddle", "RM");
+		bioValueMapping.put("rightIndex", "RI");
+		bioValueMapping.put("rightThumb", "RT");
+		bioValueMapping.put("leftEye", "LE");
+		bioValueMapping.put("rightEye", "RE");
+		Whitebox.setInternalState(exceptionBiometricsTagGenerator, "bioValueMapping", bioValueMapping);
+		Whitebox.setInternalState(exceptionBiometricsTagGenerator, "notAvailableTagValue", 
+			notAvailableTagValue);
+		List<String> regClientVersionsBeforeExceptionbiometrics = new ArrayList();
+		regClientVersionsBeforeExceptionbiometrics.add("1.1.3");
+		regClientVersionsBeforeExceptionbiometrics.add("1.1.4");
+		Whitebox.setInternalState(exceptionBiometricsTagGenerator, "regClientVersionsBeforeExceptionbiometrics",
+				regClientVersionsBeforeExceptionbiometrics);
+		Whitebox.setInternalState(exceptionBiometricsTagGenerator, "regClientCurrentVersion", regClientCurrentVersion);
+	}
+
+	@Test
+	public void testGenerateTagsForExceptionBiometricsAvailable()
+			throws BaseCheckedException, JsonMappingException, JsonProcessingException {
+		Map<String, String> metaInfoMap = new HashMap<>();
+		metaInfoMap.put(JsonConstant.EXCEPTIONBIOMETRICS, 
+				"{\"individualBiometrics\" : {\"leftEye\" : {\"type\" : \"iris\", \"missingBiometric\" : \"leftEye\",\"reason\" : \"Missing biometrics\",\"exceptionType\" : \"Permanent\",\"individualType\" : \"INDIVIDUAL\"}}}");
+		metaInfoMap.put(JsonConstant.METADATA,
+				"[ {\n  \"label\" : \"registrationId\",\n  \"value\" : \"1234\"\n},{\n  \"label\" : \"Registration Client Version Number\",\n  \"value\" : \"1.2.0.1-SNAPSHOT\"\n} ]");
+		Mockito.when(mapper.readValue(anyString(), any(Class.class)))
+				.thenReturn(new FieldValue("Registration Client Version Number", "1.2.0.1-SNAPSHOT"));

```

Comment:
Can we try to set a proper object of objectMapper instead of mapping so we can avoid the setting of client version separately again.

Reply:
fixed